src/lib/ffi/ffi_ec.cpp

```diff
@@ -0,0 +1,200 @@
+/*
+* (C) 2025 Jack Lloyd
+* (C) 2025 Dominik Schricker
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/ffi.h>
+
+#include <botan/internal/ffi_ec.h>
+#include <botan/internal/ffi_mp.h>
+#include <botan/internal/ffi_oid.h>
+#include <botan/internal/ffi_util.h>
+#include <functional>
+
+extern "C" {
+
+using namespace Botan_FFI;
+
+int botan_ec_group_destroy(botan_ec_group_t ec_group) {
+   return BOTAN_FFI_CHECKED_DELETE(ec_group);
+}
+
+int botan_ec_group_supports_application_specific_group(int* out) {
+   if(out == nullptr) {
+      return BOTAN_FFI_ERROR_NULL_POINTER;
+   }
+   if(Botan::EC_Group::supports_application_specific_group()) {
+      *out = 1;
+   } else {
+      *out = 0;
+   }
+   return BOTAN_FFI_SUCCESS;
+}
+
+int botan_ec_group_supports_named_group(const char* name, int* out) {
+   return ffi_guard_thunk(__func__, [=]() -> int {
+      if(name == nullptr || out == nullptr) {
+         return BOTAN_FFI_ERROR_NULL_POINTER;
+      }
+      if(Botan::EC_Group::supports_named_group(name)) {
+         *out = 1;
+      } else {
+         *out = 0;
+      }
+      return BOTAN_FFI_SUCCESS;
+   });
+}
+
+int botan_ec_group_from_params(botan_ec_group_t* ec_group,
+                               botan_asn1_oid_t oid,
+                               botan_mp_t p,
+                               botan_mp_t a,
+                               botan_mp_t b,
+                               botan_mp_t base_x,
+                               botan_mp_t base_y,
+                               botan_mp_t order) {
+   return ffi_guard_thunk(__func__, [=]() -> int {
+      if(ec_group == nullptr) {
+         return BOTAN_FFI_ERROR_NULL_POINTER;
+      }
+
+      Botan::EC_Group group(
+         safe_get(oid), safe_get(p), safe_get(a), safe_get(b), safe_get(base_x), safe_get(base_y), safe_get(order));
+
+      std::unique_ptr<Botan::EC_Group> group_ptr = std::make_unique<Botan::EC_Group>(std::move(group));
+      *ec_group = new botan_ec_group_struct(std::move(group_ptr));
+
+      return BOTAN_FFI_SUCCESS;
+   });
+}
+
+int botan_ec_group_from_ber(botan_ec_group_t* ec_group, const uint8_t* ber, size_t ber_len) {
+   return ffi_guard_thunk(__func__, [=]() -> int {
+      if(ec_group == nullptr || ber == nullptr) {
+         return BOTAN_FFI_ERROR_NULL_POINTER;
+      }
+
+      Botan::EC_Group group(ber, ber_len);
+
+      std::unique_ptr<Botan::EC_Group> group_ptr = std::make_unique<Botan::EC_Group>(std::move(group));
```

Comment:
Could this just be

```
auto group_ptr = std::make_unique<Botan::EC_Group>(ber, ber_len);
```

?

[In general for `x = make_unique<Foo>(...)` it's assumed the type of `x` is clear and we use `auto` rather than the explicit type]

Reply:
I guess OTOH what you have here better matches the pattern elsewhere, and the "static function named constructor" thing we do here interferes with `make_unique` in the general case...